proxy.go

```diff
@@ -1,13 +1,33 @@
 package solidproxy
 
 import (
+	"crypto/tls"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 	"time"
 )
 
-func ProxyHandler(w http.ResponseWriter, req *http.Request) {
+const (
+	MaxIdleConnections int = 20
+	RequestTimeout     int = 5
+)
+
+type Proxy struct {
+	HttpClient      *http.Client
+	HttpAgentClient *http.Client
```

Comment:
Not at a first glance. Those two objects are of the same type but they hold very different configs -- i.e. the agent one contains the client cert that is being sent during WebID-TLS.